plugins/qcheck-stm/src/ir_of_gospel.ml

```diff
@@ -167,6 +167,22 @@
   in
   ok Ir.{ formulae; modifies }
 
+let returned_value spec ret =
+  let open Tterm in
+  let is_ret vs =
+    let open Symbols in
+    Ident.equal ret vs.vs_name
+  in
+  let rec pred t =
+    match t.t_node with
+    | Tapp (ls, [ { t_node = Tvar vs; _ }; right ])
```

Comment:
```suggestion
(* match [ret = term] and return [term] *)
| Tapp (ls, [ { t_node = Tvar vs; _ }; right ])
```
since such patterns with guards are hard to decode